src/Xrefcheck/Verify.hs

```diff
@@ -782,20 +742,25 @@ checkExternalResource followed config@Config{..} link
         HttpExceptionRequest _ exc -> case exc of
           StatusCodeException resp _
             | isRedirectCode code -> case redirectLocation of
-                Nothing -> Left $ RedirectMissingLocation $ followed `pushRequest` (RedirectChainLink link)
-                Just nextLink -> case redirectRule link nextLink code ncExternalRefRedirects of
-                  Nothing -> Right RRDone
-                  Just RedirectRule{..} ->
-                    case rrOutcome of
-                      RROValid -> Right RRDone
-                      RROInvalid -> Left $ RedirectRuleError
-                        (followed `pushRequest` (RedirectChainLink link) `pushRequest` (RedirectChainLink nextLink))
-                        rrOn
-                      RROFollow -> Right $ RRFollow nextLink
-            | isAllowedErrorCode code -> Right RRDone
+                Nothing -> throwError $ RedirectMissingLocation $ followed `pushRequest` RedirectChainLink link
+                Just nextLink -> do
+                  nextUri <- ExternalResourceUriParseError `withExceptT` parseUri True nextLink
+                  nextLinkAbsolute <- case relativeTo nextUri uri of
```

Comment:
I mainly used the `relativeTo` function from the `modern-uri` package for this task. This is how we obtain the absolute target link from the (absolute) source + the (absolute or relative) target.